Documentation/auto-dependency-flow/README.md

```diff
@@ -0,0 +1,92 @@
+
+### Incremental steps toward the vision
+The narrative represents the ideal flow, and some parts won't be available in our first steps:
+
+ * The repo dependency graph is unlikely to change often enough to warrant automating its construction in the initial effort. Instead, repo dependencies will be hard-coded in the source-build repository.
+ * The CoreCLR and Standard builds may not be run in parallel.
+ * The orchestrator might not have any finalization steps. Instead, they happen during the repo builds. 
+
+## Blob feed
+The orchestrator indicates a directory on the build machine to place output files using [`/p:DotNetOutputBlobFeedDir`](api.md#pdotnetoutputblobfeeddirtarget-directory). The orchestrator makes the contents available to subsequent repo builds using [`/p:DotNetRestoreSourcePropsPath`](api.md#pdotnetrestoresourcepropspathpath) and [`/p:DotNetAssetRootUrl`](api.md#pdotnetassetrooturlurl).
+
+## Passing dependency versions
+To control the package versions a repo build uses, the orchestrator [creates](contracts.md#creation-based-on-blob-feed) and provides an easily consumable MSBuild props file. The file is placed in a directory on the build machine outside of the target repo's git repository:
+
+ * `<outside dir>/PackageVersions.props`: the [package version props](contracts.md#package-version-props) file.
+ * Additional files may be added/passed in the future, based on need. For example:
+   * Native-only repo builds may require a simpler format to reasonably parse.
+   * The more general [build output manifest](contracts.md#build-output-manifest) may be useful for advanced builds that don't use MSBuild in the same way (or at all).
+   * Additional MSBuild props files may be added to flow non-package dependencies.
+
+The orchestrator then [passes `/p:DotNetPackageVersionPropsPath=<outside dir>/PackageVersions.props`](api.md#pdotnetpackageversionpropspathpath) to every command it runs during the repo build.
+
+
+# Source build vs. product build dependency flow
+
+Source build has some requirements that don't necessarily apply to the product build: it must avoid dependencies on prebuilt dependencies and [rebuild without using the prebuilt version ("bootstrap")](https://fedoraproject.org/wiki/Packaging:Guidelines#Bootstrapping) where that's not possible. This separates the product build and source build in a few ways relevant to auto dependency flow.
+
+## Latest Target Framework Monikers (TFMs)
+E.g. upgrading to netstandard2.0, netcoreapp2.1.
+
+For product builds, we want to allow the repos continue to depend on the oldest possible TFM to give the product greater reach. Specifically, Visual Studio targets old netstandard TFMs.
+
+For source builds, upgrading TFMs allows the repos to avoid depending on old package versions. They would also need to be built from source.
```

Comment:
Do we want to spec here how this is going to be specified to the repos?  ex. `/p:DotNetSourceBuildTargetFramework=netcoreapp2.1`.

Reply:
Hmm, I'd kind of assumed that the TFM could somehow be picked up from the source-built NETStandard.Library/Microsoft.NETCore.App nupkg using a props file in the package or something. Looking a bit closer, I don't see anything like that, and it looks like CoreFX needs netcoreappX.X before Core-Setup builds, so it probably can't even work in theory.

An arg makes sense to me. It sounds like we'll need one for netstandard and one for netcoreapp.

Reply:
We can't pick up the TFM from a NuGet package because NuGet needs the TFM to know which assets to pick out of the NuGet packages.  TFMs need to be specified first - before 'restore' can take place.

>It sounds like we'll need one for netstandard and one for netcoreapp.

Do we need both?  I assumed the idea was that the whole source build would be for `netcoreapp`?  /cc @weshaggard @ericstj

Reply:
I don't have a good grasp of what this impacts during the product build itself. I imagine people using the source-build end product would want to be able to build apps against `netstandard` to have the greater targeting area, but I don't know what's necessary to make that happen.

Reply:
A really good/important question we need to figure out is: "What is the expected usage of these `source-built` assets?"

Do we expected people to reference the .nupkgs that are produced by this build?  If so, then they won't be able to build any `netstandard1.x` projects.  (Or even `netstandard2.0` when we move to a `netstandard2.1` or `3.0`.) Is this intended?

Reply:
Created https://github.com/dotnet/source-build/issues/184 to track/discuss--maybe we need some documentation around that point in particular.